community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/ReversedSingleFileTransactionCursor.java

```diff
@@ -89,14 +96,21 @@
 
         long logVersion = channel.getVersion();
         long startOffset = channel.position();
-        while ( transactionCursor.next() )
+        try
         {
-            if ( offsetCursor == offsets.length )
-            {   // Grow
-                offsets = Arrays.copyOf( offsets, offsetCursor * 2 );
+            while ( transactionCursor.next() )
+            {
+                if ( offsetCursor == offsets.length )
+                {   // Grow
+                    offsets = Arrays.copyOf( offsets, offsetCursor * 2 );
+                }
+                offsets[offsetCursor++] = startOffset;
+                startOffset = channel.position();
             }
-            offsets[offsetCursor++] = startOffset;
-            startOffset = channel.position();
+        }
+        catch ( Throwable t )
+        {
+            log.warn( buildReadErrorMessage( offsets, offsetCursor, logVersion ), t );
```

Comment:
It looks like the sketching just aborts, leaving some variables not properly initialized and still not throwing. Can't this make the upcoming reverse reading start at the wrong place?